annotators/bot_emotion_classifier/server.py

```diff
@@ -0,0 +1,416 @@
+    return decay
+
+
+def get_decay(default_mood, curr_mood):
+    decay = []
+    for i in range(len(default_mood)):
+        decay.append(get_dim_decay(default_mood[i], curr_mood[i]))
+    return decay
+
+
+# comparison of old and new mood
+def check_same_mood(curr_mood, new_mood):
+    print(curr_mood, new_mood)
+    for i in range(len(curr_mood)):
+        if curr_mood[i] * new_mood[i] < 0:
+            return False
+    return True
+
+
+# new mood calculation using difference btw current mood and current emotion
+def get_new_mood(default_mood, curr_mood, bot_emotion):
+    if check_same_mood(pad_emotions[bot_emotion], curr_mood):
+        decay = [0, 0, 0]
+    else:
+        decay = get_decay(default_mood, curr_mood)
+    vec = [pad_emotions[bot_emotion][i] - curr_mood[i] for i in range(len(curr_mood))]
+    new_mood = [0.5 * curr_mood[i] + 0.5 * vec[i] + decay[i] for i in range(len(curr_mood))]
+    new_mood_reg = [1 if dim > 1 else dim for dim in new_mood]
+    new_mood_reg = [-1 if dim < -1 else dim for dim in new_mood_reg]
+
+    # dim_symbols = [str(int(dim / abs(dim))) if dim != 0 else '-1' for dim in new_mood_reg]
+    # octant = ''.join(dim_symbols)
+    # print('New mood:', pad_moods[octant])
+    return new_mood_reg
+
+
+def get_mood_label(bot_mood):
+    octant = ""
+    for dim in bot_mood:
+        if dim == 0:
+            octant += "0"
+        elif dim > 0:
+            octant += "1"
+        else:
+            octant += "-1"
+
+    return pad_moods[octant]
+
+
+@app.route("/model", methods=["POST"])
+def respond():
+    sentences = request.json.get("sentences", [])
+    annotated_utterances = request.json.get("annotated_utterances", [])
+    bot_moods = request.json.get("bot_mood", [])
+
+    for sentence, annotated_utterance, bot_mood in zip(sentences, annotated_utterances, bot_moods):
+        user_emotion = get_emotions(annotated_utterance, probs=False)[0]
+        sentiment = get_sentiment(annotated_utterance, probs=False)[0]
+
+        logger.info("User's utterance: {}".format(sentence))
+        logger.info("User emotion: {}".format(user_emotion))
+        logger.info("Sentiment: {}".format(sentiment))
+        logger.info("Old bot mood: {}".format(bot_mood))
+
+        bot_emotion = get_bot_emotion(sentence, user_emotion, sentiment)
+        logger.info("New bot emotion: {}".format(bot_emotion))
+        print("NEW BOT EMOTION: ", bot_emotion)
+
+        new_bot_mood = get_new_mood(default_mood, bot_mood, bot_emotion)
+        logger.info("New bot mood: {}".format(new_bot_mood))
+        print("NEW BOT MOOD: ", new_bot_mood)
+
+        new_bot_mood_label = get_mood_label(new_bot_mood)
+        logger.info("New bot mood label: {}".format(new_bot_mood_label))
+        print("NEW BOT MOOD LABEL: ", new_bot_mood_label)
+
+    return jsonify(
+        [
+            {
+                "bot_mood": new_bot_mood,
+                "bot_mood_label": new_bot_mood_label,
+                "bot_emotion": bot_emotion,
```

Comment:
oops... I have noticed previously.
So, you need to create an empty list result, and append those dicts to it on every iteration of for cycle above. and then jsonify result

Reply:
so, you input and output batches would be of the same length. because now they are input of, for example, 5 samples, and output of length 1 anyway

Reply:
done